.github/workflows/retroarch.yml

```diff
@@ -0,0 +1,25 @@
+# Validates compilation of RetroArch binary
+
+name: RetroArch CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  linux-c89: # Smoketest build using most restrictive compiler and default options
+    runs-on: ubuntu-latest
+    steps:
+      - name: Dependencies
+        run: sudo apt-get install build-essential libxkbcommon-dev libx11-xcb-dev zlib1g-dev libfreetype6-dev libegl1-mesa-dev libgles2-mesa-dev libgbm-dev nvidia-cg-toolkit nvidia-cg-dev libavcodec-dev libsdl2-dev libsdl-image1.2-dev libxml2-dev yasm
```

Comment:
This is the set of items described here: https://docs.libretro.com/development/retroarch/compilation/linux-and-bsd/#satisfying-dependencies-under-debianubuntu, plus libx11-xcb-dev